homeassistant/components/shelly/__init__.py

```diff
@@ -143,6 +143,8 @@ def __init__(self, hass, entry, device: aioshelly.Device):
         )
         self._last_input_events_count = dict()
 
+        hass.bus.async_listen(EVENT_HOMEASSISTANT_STOP, self._handle_ha_stop)
```

Comment:
We should unsubscribe the listener when removing the entry. Also probably use `async_listen_once`.

Reply:
`async_listen_once` still shows the original issues from time to time